docs/topics/gradle/gradle-plugin-variants.md

```diff
@@ -0,0 +1,93 @@
+[//]: # (title: Support for Gradle plugin variants)
+
+Gradle 7.0 introduced a new feature for Gradle plugin authors
+— [plugins with variants](https://docs.gradle.org/7.0/userguide/implementing_gradle_plugins.html#plugin-with-variants).
+This feature makes it easier to add support for new Gradle features while maintaining compatibility for Gradle versions
+below 7.1. Learn more about [variant selection in Gradle](https://docs.gradle.org/current/userguide/variant_model.html).
+
+With Gradle plugin variants, the Kotlin team can ship different Kotlin Gradle plugin (KGP) variants for different Gradle versions. 
+The goal is to support the base Kotlin compilation in the `main` variant, which corresponds to the oldest supported versions of
+Gradle. Each variant will have implementations for Gradle features from a corresponding release. The latest variant will
+support the widest Gradle feature set. With this approach, it is possible to extend support for older Gradle versions 
+with limited functionality.
+
+Currently, there are only two variants of the Kotlin Gradle plugin:
+
+* `main` for Gradle versions 6.7.1–6.9.2
+* `gradle70` for Gradle versions 7.0 and higher
```

Comment:
It is more in Kotlin 1.8.0:
- `gradle71` for Gradle 7.1 - 7.4
- `gradle75` for Gradle 7.5
- `gradle76` for Gradle 7.6 and higher